nixos/lib/test-driver/src/test_driver/machine/__init__.py

```diff
@@ -919,17 +849,17 @@ def screenshot(self, filename: str) -> None:
             filename += ".png"
         if "/" not in filename:
             filename = os.path.join(self.out_dir, filename)
-        tmp = f"{filename}.ppm"
 
         with self.nested(
             f"making screenshot {filename}",
             {"image": os.path.basename(filename)},
         ):
-            self.send_monitor_command(f"screendump {tmp}")
-            ret = subprocess.run(f"pnmtopng '{tmp}' > '{filename}'", shell=True)
-            os.unlink(tmp)
-            if ret.returncode != 0:
-                raise MachineError("Cannot convert screenshot")
+            with self._managed_screenshot() as screenshot_path:
+                ret = subprocess.run(
+                    f"pnmtopng '{screenshot_path}' > '{filename}'", shell=True
+                )
+                if ret.returncode != 0:
+                    raise MachineError("Cannot convert screenshot")
```

Comment:
Can we provide more context into what went wrong? Does `ret` have access to `stdout` and/or `stderr`? At the minimum, we could include the returncode?

```
raise MachineError(f"Converting screenshot returned a non-zero status: {ret.returncode}")
```